camel/agents/role_playing.py

```diff
@@ -266,17 +266,15 @@ def step(
             whether or not the user agent terminated the conversation, and
             any additional user information.
         """
-        user_response = self.user_agent.step(
-            assistant_msg.to_user_chat_message())
+        user_response = self.user_agent.step(assistant_msg)
         if user_response.terminated or user_response.msgs is None:
             return (ChatAgentResponse([], False, {}),
                     ChatAgentResponse([], user_response.terminated,
                                       user_response.info))
         user_msg = self.process_messages(user_response.msgs)
         self.user_agent.update_messages(user_msg)
 
-        assistant_response = self.assistant_agent.step(
-            user_msg.to_user_chat_message())
+        assistant_response = self.assistant_agent.step(user_msg)
```

Comment:
How does this work as expected? We need to set `role` to `'user'` right?

Reply:
user_msg is UserChatMessage and step takes ChatMessage as input. A derived class can be always cast to the base class. OOO baby!

Reply:
@lightaime what was the original purpose of
```
self.assistant_agent.step(
            user_msg.to_user_chat_message())
```
Convert a user message to a user massage?... user_msg is a CatMessage anyway